Classic Intel compilers do not support gcc-toolchain

stephenmsachs opened the pull request:
This fix removes `--gcc-toolchain=` from the ~.fcg` files for the classic Intel compilers. AFAIK this option is only supported for Clang based compilers.

This lead to an issue when installing cmake. Reproducer:
```
spack install cmake@3.24.2%intel@2021.7.0~doc+ncurses+ownlibs~qt
build_type=Release arch=linux-amzn2-skylake_avx512
```

Tagging maintainer @rscohn2

Commit:
Classic Intel compilers do not support gcc-toolchain

This fix removes `--gcc-toolchain=` from the ~.fcg` files for the classic Intel
compilers. AFAIK this option is only supported for Clang based compilers.

This lead to an issue when installing cmake. Reproducer:
```
spack install cmake@3.24.2%intel@2021.7.0~doc+ncurses+ownlibs~qt
build_type=Release arch=linux-amzn2-skylake_avx512
```

Tagging maintainer @rscohn2

rscohn2 commented:
@haampie @skosukhin:

I see more problems now.

icc accepts gcc-name, icx accepts -gcc-toolchain
icc does not accept -Wno-unused-command-line-argument, but does not need it because it does not warn against Wl,-rpath and -c

I will refactor this a little more and test against all compilers.

haampie commented:
Oof, thanks for doing this!

Commit:
Add `-gcc-name` for icc

.. and `-gxx-name` for icpc.

AFAIK this is used for modern C++ support, so we can ignore `ifort`.

stephenmsachs committed:
Merge branch 'develop' into intel-classic-no-gcc-toolchain

rscohn2 reviewed (approved):
Thanks for the fix.
